### PR TITLE
M2C-21768 Validate product quantity on Wishlist update

### PR DESCRIPTION
**Pull request by ptylek**
### Description (*)

I've added a method which validates product qty in order to prevent updating Wishlist with product quantity that is empty, 0 or negative number.

### Fixed Issues (if relevant)

1. magento/magento2#25032: Display some error "We can't update your Wish List right now." at wish list

### Manual testing scenarios (*)

1. Add product to Wishlist
2. Go to Wishlist
3. Edit the product from Wishlist
4. On detail page leave qty blank, add 0 or negative number and click on Update Wishlist button.
5. Error message should appear.

image

### Questions or comments

The same behaviour works for updating products quantity in Wishlist in User Area. 

image

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by krzksz — CHANGES_REQUESTED** [review]
Hey @ptylek, thanks for the contribution! Please check the tests results and fix the code accordingly.

File: app/code/Magento/Wishlist/view/frontend/web/js/add-to-wishlist.js [hunk 1]
```diff
@@ -30,8 +31,10 @@ define([
         _bind: function () {
             var options = this.options,
                 dataUpdateFunc = '_updateWishlistData',
+                validateProductQty = '_validateWishlistQty',
                 changeCustomOption = 'change ' + options.customOptionsInfo,
                 changeQty = 'change ' + options.qtyInfo,
+                updateWishlist = 'click ' + options.actionElement,
```
Comment (Karlasa) [inline app/code/Magento/Wishlist/view/frontend/web/js/add-to-wishlist.js:37]: failed test https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/25641/0a4e4127615ba161e7e73f26a6e16db6/Statics/allure-report-ce/index.html#suites/5533c1ade71e7287a27c7a2c90cb7724/85783273d3bd5f09/

Error - 'updateWishlist' is assigned a value but never used. Allowed unused vars must match /config/. (no-unused-vars)

Cheers, 
Karla

**Review by Karlasa — CHANGES_REQUESTED** [review]

File: app/code/Magento/Wishlist/view/frontend/web/js/add-to-wishlist.js (hunk 1 above)
Comment (ptylek) [reply app/code/Magento/Wishlist/view/frontend/web/js/add-to-wishlist.js:37]: Hi! Thanks for help. This var is used in line 51, just like any other event.

**Comment by krzksz** [comment]
@magento run static tests

**Comment by krzksz** [comment]
@magento run all tests

File: app/code/Magento/Wishlist/view/frontend/web/js/add-to-wishlist.js (hunk 1 above)
Comment (Karlasa) [reply app/code/Magento/Wishlist/view/frontend/web/js/add-to-wishlist.js:37]: Yeah, seems like some issue with tests :)

**Comment by Karlasa** [comment]
will re-open PR to see if it fixes the issue with failed test.

**Comment by ptylek** [comment]
@Karlasa I tried to close and reopen it once again, without success.

**Review by Karlasa — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @Karlasa, thank you for the review.
[ENGCOM-6305](https://jira.corp.magento.com/browse/ENGCOM-6305) has been created to process this Pull Request
:eight_spoked_asterisk: @Karlasa, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Review by krzksz — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @krzksz, thank you for the review.
[ENGCOM-6305](https://jira.corp.magento.com/browse/ENGCOM-6305) has been created to process this Pull Request
:eight_spoked_asterisk: @krzksz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
Hi @ptylek .
     Due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all code must be covered by tests. Please cover your fix by automated tests.
Thanks!

**Comment by Karlasa** [comment]
@engcom-Alfa Could you provide some insights why static tests are failing without giving any feedback about it?

**Comment by engcom-Alfa** [comment]
@Karlasa There are many static errors in the `add-to-wishlist.js` file. 
The errors are not displayed because they do not fit into the report frame.

**Comment by ptylek** [comment]
@engcom-Alfa I fixed those error. All checks have passed.

**Comment by engcom-Golf** [comment]
I will take care of test coverage

**Comment by engcom-Golf** [comment]
Hi @krzksz changes covered by jasmine test, could you please review ?

**Review by krzksz — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @krzksz, thank you for the review.
[ENGCOM-6305](https://jira.corp.magento.com/browse/ENGCOM-6305) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**